packages/cloudflare/src/api/do-sharded-tag-cache.ts

```diff
@@ -130,18 +208,39 @@ class ShardedD1TagCache implements NextModeTagCache {
   async writeTags(tags: string[]): Promise<void> {
     const { isDisabled } = await this.getConfig();
     if (isDisabled) return;
-    const shards = this.generateShards(tags);
+    const shards = this.generateShards(tags, true);
+    const currentTime = Date.now();
     // We then create a new durable object for each shard
     await Promise.all(
       Array.from(shards.entries()).map(async ([shardId, shardedTags]) => {
-        const stub = this.getDurableObjectStub(shardId);
-        await stub.writeTags(shardedTags);
-        // Depending on the shards and the tags, deleting from the regional cache will not work for every tag
-        await this.deleteRegionalCache(shardId, shardedTags);
+        await this.performWriteTagsWithRetry(shardId, shardedTags, currentTime);
       })
     );
   }
 
+  async performWriteTagsWithRetry(shardId: string, tags: string[], lastModified: number, retryNumber = 0) {
+    if (retryNumber >= this.maxWriteRetries) {
```

Comment:
nit: this could be moved l240 to be closer to the related logic (it would also avoid an extra-call but not the main reason)